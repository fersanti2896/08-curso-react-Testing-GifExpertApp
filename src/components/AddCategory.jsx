import { useState } from "react"

export const AddCategory = ( { newCategory } ) => {
    const [inputValue, setInputValue] = useState('');
    
    /**
     * Función que obtiene el valor del input.
     * @param {event} event.target 
     */
    const onInputChange = ({ target }) => {
        setInputValue(target.value);
    }

    /**
     * Función que agrega el valor del input al arreglo de categorías.
     * @param {event} event 
     * @returns 
     */
    const onSubmit = (event) => {
        event.preventDefault();

        if( inputValue.trim().length <= 1 ) return;

        newCategory(inputValue.trim());
        setInputValue('');
    }

    return (
        <>
            <form onSubmit={ (event) => onSubmit(event) } >
                <input type="text"
                    placeholder="Buscar Gifs"
                    value={ inputValue }
                    onChange={ onInputChange }
                />
            </form>
        </>
    )
}
