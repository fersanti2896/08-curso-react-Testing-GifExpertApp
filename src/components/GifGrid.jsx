import PropTypes from 'prop-types';

import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category }) => {
    const { images, isLoading } = useFetchGifs( category );
    
    return (
        <>
            <h5>{ category }</h5>
            { isLoading && (<h5>Cargando...</h5>) }

            <div className='card-grid'>
                { 
                    images.map( (image) => ( 
                        <GifGridItem key={ image.id } 
                                     title={ image.title }
                                     url={ image.url } /> 
                    ) ) 
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}

GifGrid.defaultProps = {
    category: 'Shrek'
}
