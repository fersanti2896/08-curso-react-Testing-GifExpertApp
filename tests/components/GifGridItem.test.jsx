import { render, screen } from "@testing-library/react";
import { GifGridItem } from "../../src/components/GifGridItem";

describe('Pruebas en <GifGridItem />', () => {
    const title = 'El gato con Botas GIF';
    const url = 'https://giphy.com/gifs/shrek-peacocktv-TIGP3k4gNAqvza2KJK';
  

    test('Debe hacer match con el snapshot.', () => {
        const { container } = render(<GifGridItem title={ title } url={url} />);

        expect( container ).toMatchSnapshot();
    });

    test('Debe mostrar el title "El gato con Botas GIF".', () => {
        render(<GifGridItem title={title} url={ url }/>);

        expect( screen.getByText(title) ).toBeTruthy();
    });

    test('Debe mostrar la imagen con la URL y el ALT indicado.', () => {
        render(<GifGridItem title={ title } url={ url }/>);
        
        const { src, alt } = screen.getByRole('img');
        
        expect( src ).toContain( url );
        expect( alt ).toContain( title);
    });
});