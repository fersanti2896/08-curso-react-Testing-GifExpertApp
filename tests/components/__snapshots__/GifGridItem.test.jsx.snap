// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe hacer match con el snapshot. 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="El gato con Botas GIF"
      src="https://giphy.com/gifs/shrek-peacocktv-TIGP3k4gNAqvza2KJK"
    />
    <p>
      El gato con Botas GIF
    </p>
  </div>
</div>
`;
